tests/test_service_v2.py

```diff
@@ -1563,6 +1606,77 @@ def test_batch_request(service):
     assert len(chset_response) == 1
     assert chset_response[0] is None   # response to update request is None
 
+@responses.activate
+def test_batch_request_with_decoded_path(service):
+    """Batch requests"""
+
+    # pylint: disable=redefined-outer-name
+
+    response_body = (b'--batch_r1\n'
+                     b'Content-Type: application/http\n'
+                     b'Content-Transfer-Encoding: binary\n'
+                     b'\n'
+                     b'HTTP/1.1 200 OK\n'
+                     b'Content-Type: application/json\n'
+                     b'\n'
+                     b'{"d": {"ID": 23, "NameFirst": "Rob", "NameLast": "Ickes", "Address": { "ID": 456, "Street": "Baker Street", "City": "London"} }}'
+                     b'\n'
+                     b'--batch_r1\n'
+                     b'Content-Type: multipart/mixed; boundary=changeset_1\n'
+                     b'\n'
+                     b'--changeset_1\n'
+                     b'Content-Type: application/http\n'
+                     b'Content-Transfer-Encoding: binary\n'
+                     b'\n'
+                     b'HTTP/1.1 204 Updated\n'
+                     b'Content-Type: application/json\n'
+                     b'\n'
+                     b"{b'd': {'Sensor': 'Sensor-address', 'Date': datetime\'2017-12-24T18:00\', 'Value': '34.0d'}}"
+                     b'\n'
+                     b'--changeset_1--\n'
+                     b'\n'
+                     b'--batch_r1--')
+
+    responses.add(
+        responses.POST,
+        f'{URL_ROOT}/$batch',
+        body=response_body,
+        content_type='multipart/mixed; boundary=batch_r1',
+        status=202)
+
+    batch = service.create_batch('batch1')
+
+    chset = service.create_changeset('chset1')
+
+    employee_request = service.entity_sets.Employees.get_entity(23, encode_path=False)
```

Comment:
I was able to find only `encode_path=False` scenario for a batch payload. I would expect at least some encode_path=False scenarios for direct, non-batch URLs as well, all other tests used the default True and wrap the resource path vie `quote()`. Does not need to be copy paste of every modified test , but perhaps adding of one new test - ideallz parametrized, with various problematic resource paths based on different types (string, date etc.)

Reply:
There is one test which tests delete and one for get requests with not encoding the path.
[test_delete_entity_not_encoded_path](https://github.com/linda-sap/python-pyodata/blob/1e6e1a982ea6ee8b16d32058526d7fc7fd7d11c5/tests/test_service_v2.py#L661)
[test_get_entity_not_encoded_path](https://github.com/linda-sap/python-pyodata/blob/1e6e1a982ea6ee8b16d32058526d7fc7fd7d11c5/tests/test_service_v2.py#L1478)

Is this sufficient for you?